components/announcement_bar/enable_notifications_bar/enable_notifications_bar.tsx

```diff
@@ -0,0 +1,65 @@
+// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
+// See LICENSE.txt for license information.
+
+import React, {useEffect, useState} from 'react';
+
+import {AnnouncementBarTypes} from 'utils/constants';
+
+import {t} from 'utils/i18n';
+import AnnouncementBar from '../default_announcement_bar';
+
+type Props = {
+    show: boolean;
+    actions: {
+        enableBrowserNotifications: () => void;
+        trackEnableNotificationsBarDisplay: () => void;
+        disableNotificationsPermissionRequests: () => void;
+    };
+}
+
+const EnableNotificationsBar = ({show, actions}: Props) => {
+    const [isDisplayed, setIsDisplayed] = useState(false);
+
+    useEffect(() => {
+        setIsDisplayed(show);
+    }, [show]);
+
+    useEffect(() => {
+        if (isDisplayed) {
+            actions.trackEnableNotificationsBarDisplay();
+        }
+    }, [isDisplayed, actions]);
+
+    if (!isDisplayed) {
+        return null;
+    }
+
+    const handleEnableButtonClick = () => {
```

Comment:
Could you wrap these in `useCallback` so that we're not passing new functions into `AnnouncementBar` each time this component renders?

Reply:
Done.